sharing/src/test/scala/io/delta/sharing/spark/TestClientForDeltaFormatSharing.scala

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.sharing.spark
+
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.sql.delta.util.JsonUtils
+import io.delta.sharing.client.{
+  DeltaSharingClient,
+  DeltaSharingProfileProvider,
+  DeltaSharingRestClient
+}
+import io.delta.sharing.client.model.{
+  AddFile => ClientAddFile,
+  DeltaTableFiles,
+  DeltaTableMetadata,
+  SingleAction,
+  Table
+}
+
+import org.apache.spark.SparkEnv
+import org.apache.spark.storage.BlockId
+
+/**
+ * A mocked delta sharing client for DeltaFormatSharing
```

Comment:
Can you add bit more details on how this can be used with an example?

Reply:
added